packages/input-option/src/InputOption/InputOption.types.ts

```diff
@@ -23,6 +23,21 @@ type RenderedContext = (typeof RenderedContext)[keyof typeof RenderedContext];
 
 export { RenderedContext };
 
+const State = {
+  Default: 'default',
+  Hover: 'hover',
+  Highlight: 'highlight',
+  Disabled: 'disabled',
+  Checked: 'checked',
+  Destructive: 'destructive',
+} as const;
```

Comment:
What's the purpose of this type? It doesn't look like we're exposing it to consumers, and the fact that the states in this type aren't actually mutually exclusive is confusing (e.g. an option can be both "hovered" and "checked" at the same time, so what is `State` in this case?)

Reply:
To me there are multiple different "state"-type variables:

`type State =  "default" | "disabled" | "checked" | "highlight"`

`type ActionType = "default" | "destructive"`

and also `isHovered` (which can apply to any combination of `State` & `ActionType`

Reply:
Then, in `.style.ts` we have 

`get{Element}StateStyle(context)`, `get{Element}ActionTypeStyle(context)` and `get{Element}HoverStyle(context)`

Reply:
You'd also probably also want to simplify `XYZContextColors` to match